lua/pixelui/core/cl_images.lua

```diff
@@ -50,15 +50,27 @@ local function processQueue()
     end
 end
 
-function PIXEL.GetImgur(id, callback, _, matSettings)
-    if materials[id] then
-        callback(materials[id])
-    elseif file.Exists("pixel/" .. id .. ".png", "DATA") then
-        materials[id] = Material("../data/pixel/" .. id .. ".png", matSettings or "noclamp smooth mips")
-        callback(materials[id])
+function PIXEL.GetImage(url, callback, matSettings)
+    local urlWithoutProtocol = string.gsub(url, "https://", "")
+    urlWithoutProtocol = string.gsub(urlWithoutProtocol, "http://", "")
+
+    local fileName = url:match("[^/]+$")
```

Comment:
Instead of matching this pattern twice, find the string first and get the file name. Eg:
```lua
local fileNameStart, fileNameEnd = url:find("[^/]+$")
local fileName = url:sub(fileNameStart, fileNameEnd)
```
It is important to note that this will error when the file name cannot be found as string.find returns nil so make sure you handle this case.